example5.ts

```diff
@@ -0,0 +1,80 @@
+import * as readline from 'readline';
+
+type Position = {
+  x: number;
+  y: number;
+};
+
+const maze: string[][] = [
+  [' ', ' ', ' ', '*', ' ', ' ', ' '],
+  ['*', '*', ' ', '*', ' ', '*', ' '],
+  [' ', ' ', ' ', ' ', ' ', ' ', ' '],
+  [' ', '*', '*', '*', '*', '*', ' '],
+  [' ', ' ', ' ', ' ', ' ', ' ', 'E'], // Exit
+];
+
+let playerPosition: Position = { x: 0, y: 0 }; // Starting position
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+function printMaze(): void {
+  maze.forEach((row, y) => {
+    let rowStr: string = '';
+    row.forEach((cell, x) => {
```

Comment:
Consider extracting the logic inside the forEach loop into a separate function for better readability and maintainability. It is your destiny.